classes/converter.php

```diff
@@ -166,10 +172,10 @@ public function start_document_conversion(\core_files\conversion $conversion) {
         $filepath = $localpath ?? $remotepath;
         $type = '';
         $filename = $file->get_filename();
-        $contenthashf7 = substr($contenthash, 0, 7);
-        $data = array('file' => curl_file_create($filepath, $type, $contenthashf7.$filename));
-
+        $data = array('file' => curl_file_create($filepath, $type, $contenthash.$filename));
```

Comment:
Maybe instead of rely on ``$filename``. We could maybe use the combination of ``contenthash`` and ``pathnamehash``. Because of the limitation of filename length in the OS (Max 255 bytes).